hydra_base/lib/project.py

```diff
@@ -125,19 +131,34 @@ def update_project(project, **kwargs):
 
     proj_i = _get_project(project.id, user_id, check_write=True)
 
-    proj_i.name = project.name
-    proj_i.description = project.description
+    if project.name:
+        proj_i.name = project.name
+
+    if project.description:
+        proj_i.description = project.description
+    
+
+    if project.appdata is not None:
+        if proj_i.appdata is None:
+            proj_i.appdata = dict(project.appdata)
+        else:
+            newdict = dict(proj_i.appdata)
+            for k, v in project.appdata.items():
+                newdict[k] = v
+            proj_i.appdata = newdict
```

Comment:
`dict.copy()` creates a new instance and `dict_a.update(dict_b)` adds `dict_b` elements to `dict_a`, overwriting duplicates.